plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -153,6 +153,27 @@ internal fun Package.toSpdxPackage(
 
     val resolvedLicenseExpressions = licenseInfoResolver.resolveLicenseInfo(id).filterExcluded()
 
+    val declaredPackageLicenses = resolvedLicenseExpressions.filter(LicenseView.ONLY_DECLARED)
+
+    val matcher = PathLicenseMatcher(LicenseFilePatterns.getInstance())
+    val licensePaths = resolvedLicenseExpressions.flatMap { licenseInfo ->
+        licenseInfo.locations.map { it.location.path }
+    }
+
+    val packageRootPath = vcsProcessed.path.takeIf { type == SpdxPackageType.VCS_PACKAGE }.orEmpty()
+    val applicableLicensePaths = matcher.getApplicableLicenseFilesForDirectories(licensePaths, listOf(packageRootPath))
+    val applicableLicenseFiles = applicableLicensePaths[packageRootPath].orEmpty()
+
+    val detectedPackageLicenses = resolvedLicenseExpressions.filterTo(mutableSetOf()) { licenseInfo ->
+        licenseInfo.locations.any {
+            it.location.path in applicableLicenseFiles
+        }
+    }
+
+    val packageLicenseExpressions = (declaredPackageLicenses.licenses + detectedPackageLicenses)
+        .flatMap { it.originalExpressions }
+        .map { it.expression }
```

Comment:
Can you help me understand why `originalExpressions` are used, or rather what these are?

Reply:
I basically was finding this out only by experiment myself. I was first using `license`, but that's a `SpdxSingleLicenseExpression`. In order to maintain compound expressions, `originalExpressions` has to be used, see

https://github.com/oss-review-toolkit/ort/blob/06059ddd268a9e58c2aaab203ed293f47aa94d3b/model/src/main/kotlin/licenses/ResolvedLicense.kt#L44-L47

Reply:
Interresting, thanks!